numpy/ma/core.py

```diff
@@ -1481,9 +1481,14 @@ def getmaskarray(arr):
     """
     mask = getmask(arr)
     if mask is nomask:
-        mask = make_mask_none(np.shape(arr), getattr(arr, 'dtype', None))
+        mask = make_mask_none(
+            np.shape(arr),
+            dtype=getattr(arr, 'dtype', None),
+            readonly=nomask_as_readonly)
     return mask
 
+def _viewmaskarray(arr):
```

Comment:
I had actually not thought so much about the flags, but more that `writable=True` indicated the array had to be writable. Maybe we can be even more direct and make it `broadcast_if_possible=True`?